### ISSUE TITLE
Can someone ELI5 how to install locally with docker? Docs are vague.

### ISSUE THREAD

**Issue by EmmaWebGH** [Mintplex-Labs/anything-llm#479]
OK Look I don't use docker much, sorry. I am a python/venv person. Docker does my head in.

What I did:
`docker pull mintplexlabs/anythingllm:master`
Then I can see it in Docker Desktop for Windows.

Made a dir and downloaded all the docker files into it, including `docker-compose.yml`
All files from here:
https://github.com/Mintplex-Labs/anything-llm/tree/master/docker
Open cmd, cd to the dir with the yml
Then run `docker compose up`

But
It errors out saying it can't find files.
```
 ! anything-llm Warning                                                                                            6.0s
[+] Building 0.0s (0/0)                                                                                  docker:default
could not find C:\AI\Text\docker: CreateFile C:\AI\Text\docker: The system cannot find the file specified.
```

Contrast this to my docker experience with Cheshire Cat for LLMs, where it actually just works.

I get also that you probably really want to funnel more people to the paid hosted version and not just have us all run it locally free, but I gotta say there's a serious lack of information on how to get this thing running. None of the Youtube videos show it - whether from Mintplex or any 3rd party people, and I find this really odd. It's it's such a king app why aren't more people making vids about it?

I'd really, REALLY like to try it though as I've been on a quest for decent local LLM+RAG + configurability for weeks now and can't find anything that really works well yet.

**Comment by lunamidori5** [comment]
download the full repo then go into the docker folder then do ``docker compose up -d``

**Comment by timothycarambat** [comment]
Docker docs now contain a copy/paste snippet you can run in Powershell in Windows that will mount an external storage folder in the Powershell users `Documents` folder so you can have persistent storage without downloading the repo

https://github.com/Mintplex-Labs/anything-llm/blob/master/docker/HOW_TO_USE_DOCKER.md#recommend-way-to-run-dockerized-anythingllm

**Comment by EmmaWebGH** [comment]
> Docker docs now contain a copy/paste snippet you can run in Powershell in Windows that will mount an external storage folder in the Powershell users `Documents` folder so you can have persistent storage without downloading the repo
> 
> https://github.com/Mintplex-Labs/anything-llm/blob/master/docker/HOW_TO_USE_DOCKER.md#recommend-way-to-run-dockerized-anythingllm

Thanks for this. I didn't realise I had to git-pull, I thought running the docker command pulled everything into docker, similar to a git pull. Thanks kindly for the help! Can't wait to get this going and I'll try now.

**Comment by EmmaWebGH** [comment]
Ok, wait, I'm confused again. **Clone** the full repo then go into docker folder and `docker compose up -d.`
Or **DON'T clone the repo**, and instead use that snipped _"so you can have persistent storage without downloading the repo"._
What's preferred?

Can I do this
`$env:STORAGE_LOCATION="C:\AI\Text\anything-llm\storage"; `

or is it not recommended to have the storage folder inside my main (git cloned) anything-llm folder? Will that cause problems when I update in the future?

I prefer to keep all things to do with an LLM app, in one location. Otherwise if I ever remove it, I have to remember that anything-llm has a bunch of stuff in my docs/home folder, which I would prefer not to be treating as a dir full of app files.

Also I've found docs in two locations, which should we be looking at?
`https://github.com/Mintplex-Labs/anything-llm/blob/master/docker/HOW_TO_USE_DOCKER.md#recommend-way-to-run-dockerized-anythingllm`
or
`https://docs.useanything.com/private-deployment/run-anythingllm-locally`

Thanks kindly

**Comment by timothycarambat** [comment]
@EmmaWebGH Since its docker and its open source, there are about 100 different ways you could run AnythingLLM. Since you are on windows I would do this:

in Powershell/terminal:
`docker pull mintplexlabs/anythingllm:master` to get the latest image.

Then You can create a folder _anywhere_ on your host machine. Wherever you want to keep all your AnythingLLM information. If `C:\AI\Text\anything-llm\storage` is a real location on your drive, sure. It should ideally be an empty folder, but it doesn't matter.

Then you can run this command in Powershell:
```
$env:STORAGE_LOCATION="C:\AI\Text\anything-llm\storage"; `
If(!(Test-Path $env:STORAGE_LOCATION)) {New-Item $env:STORAGE_LOCATION -ItemType Directory}; `
If(!(Test-Path "$env:STORAGE_LOCATION\.env")) {New-Item "$env:STORAGE_LOCATION\.env"}; `
docker run -d -p 3001:3001 `
--cap-add SYS_ADMIN `
-v "$env:STORAGE_LOCATION`:/app/server/storage" `
-v "$env:STORAGE_LOCATION\.env:/app/server/.env" `
-e STORAGE_DIR="/app/server/storage" `
mintplexlabs/anythingllm:master;
```

Now you can visit `localhost:3001` and use AnythingLLM. Any data, embeddings, and everything will be saved to `C:\AI\Text\anything-llm\storage` and if you ever want to upgrade your docker container, you can delete is from docker, then re-run the `docker pull` command and the Powershell command again, and you will have the latest code running but still have your data.


You can do exactly as @lunamidori5 did as well and it will work the same! There is no clear benefit to running either over the other. Its simply just another way to do it.

**Comment by lunamidori5** [comment]
The benefit of running it the way I do. Is you only have to remember three words. In order to reinstall / restart the container. It is also persistent. And you don't have to reinstall it every time you wish to run it. But my apologies for causing confusion!
